### PR TITLE
Component/tag

### PR DESCRIPTION
**Pull request by gurjit03**
Small-sized
image

Large-sized
image

### CODE REVIEW

File: packages/open-ui/.vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (alexpricedev) [inline packages/open-ui/.vscode/settings.json:1]: Can you pls remove this and add .vscode to the `.gitignore`

File: packages/open-ui/src/components/Button/index.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+"use strict";
```
Comment (alexpricedev) [inline packages/open-ui/src/components/Button/index.js]: Pls remove this. It shouldnt be here!

File: packages/open-ui/src/components/Tag/Tag.module.scss [hunk 3]
```diff
@@ -0,0 +1,16 @@
+@import 'styles/index.scss';
+
+.Tag {
```
Comment (alexpricedev) [inline packages/open-ui/src/components/Tag/Tag.module.scss]: I think we're keeping everything `lowerCamelCase` for classes

File: packages/open-ui/src/components/Tag/Tag.module.scss [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  background-color: $grey;
+  border-radius: 12px;
+  color: $black;
+  display: inline-block;
+  font-size: $font-size-sm;
+  padding: $vr;
```
Comment (alexpricedev) [inline packages/open-ui/src/components/Tag/Tag.module.scss]: I'd like to see a little more horizontal padding I think maybe `2 *` on the the Y-axis

File: packages/open-ui/src/components/Tag/Tag.module.scss [hunk 5]
```diff
@@ -0,0 +1,16 @@
+@import 'styles/index.scss';
+
+.Tag {
+  background-color: $grey;
+  border-radius: 12px;
+  color: $black;
+  display: inline-block;
+  font-size: $font-size-sm;
```
Comment (alexpricedev) [inline packages/open-ui/src/components/Tag/Tag.module.scss:8]: Did you try `font-weight: 500`?

File: packages/open-ui/src/components/Tag/Tag.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+
+import { Tag } from './Tag';
+
+export const standard = () => (
+  <React.Fragment>
+    <Tag name="tag1" />
```
Comment (alexpricedev) [inline packages/open-ui/src/components/Tag/Tag.stories.tsx]: I think `text=` would be a clearer API? Whatcha think?

File: packages/open-ui/src/components/Tag/Tag.module.scss [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  padding: $vr;
+  text-transform: uppercase;
+}
+.large {
+  padding: $vr * 1.5;
+  border-radius: 20px;
```
Comment (alexpricedev) [inline packages/open-ui/src/components/Tag/Tag.module.scss]: For rounded corners pls use `border-radius: 50%`;

File: packages/open-ui/src/components/Tag/Tag.tsx [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+import styles from './Tag.module.scss';
+
+interface TagProps {
+  name: string;
+  size?: 'small' | 'large';
+}
+
+export const Tag: React.FC<TagProps & MarginStyleKeys> = props => {
+  const { name, size = 'small' } = props;
+  const _props = { marginBottom: 0.5, marginRight: 0.5, ...props };
```
Comment (alexpricedev) [inline packages/open-ui/src/components/Tag/Tag.tsx]: I dont think there should be any default margins on *any* components. Lets keep it clean so we can use them as we need in `client`

File: packages/open-ui/src/components/Tag/Tag.tsx [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+import styles from './Tag.module.scss';
+
+interface TagProps {
+  name: string;
+  size?: 'small' | 'large';
+}
+
+export const Tag: React.FC<TagProps & MarginStyleKeys> = props => {
+  const { name, size = 'small' } = props;
```
Comment (alexpricedev) [inline packages/open-ui/src/components/Tag/Tag.tsx]: I dont think we need to set a default here as `small` is the default style anyway

File: packages/open-ui/src/components/Tag/Tag.tsx [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+  const _props = { marginBottom: 0.5, marginRight: 0.5, ...props };
+  const marginStyles = getMarginProps(_props);
+  const cls = classNames(styles.Tag, {
+    [styles.large]: size === 'large',
+  });
+  return (
+    <div style={{ ...marginStyles }} className={cls}>
```
Comment (alexpricedev) [inline packages/open-ui/src/components/Tag/Tag.tsx]: You can just pass `{marginStyles}` i think

File: packages/open-ui/src/main.js [hunk 11]
```diff
@@ -0,0 +1,4 @@
+"use strict";
```
Comment (alexpricedev) [inline packages/open-ui/src/main.js]: This shouldnt be here i dont think. Looks like some old export code

**Review by alexpricedev — CHANGES_REQUESTED** [review]

File: packages/open-ui/src/components/Tag/Tag.module.scss (hunk 7 above)
Comment (gurjit03) [reply packages/open-ui/src/components/Tag/Tag.module.scss]: @remotealex I already tried it I believe, they were creating too much of roundness

File: packages/open-ui/src/components/Tag/Tag.tsx (hunk 8 above)
Comment (gurjit03) [reply packages/open-ui/src/components/Tag/Tag.tsx]: @remotealex the only reason I put em was to not include the marginStyles everytime we gonna use them as I felt we gonna use them in form of list. But yeah I agree to not have default margins

File: packages/open-ui/.vscode/settings.json (hunk 1 above)
Comment (alexpricedev) [reply packages/open-ui/.vscode/settings.json:1]: I think you need to remove this still. You can use `git rm -r --cached` i think

File: packages/open-ui/src/components/Tag/Tag.tsx [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+import styles from './Tag.module.scss';
+
+interface TagProps {
+  text: string;
+  size?: 'large';
+}
+
+export const Tag: React.FC<TagProps & MarginStyleKeys> = props => {
+  const { text, size, ...marginStyles } = props;
+  const _marginStyles = getMarginProps(marginStyles);
```
Comment (alexpricedev) [inline packages/open-ui/src/components/Tag/Tag.tsx:16]: You can just pass props in here:
```
  const marginProps = getMarginProps(props);
```

File: packages/open-ui/src/components/Tag/Tag.tsx [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+  const cls = classNames(styles.tag, {
+    [styles.large]: size === 'large',
+  });
+  return (
+    <div style={{ ..._marginStyles }} className={cls}>
```
Comment (alexpricedev) [inline packages/open-ui/src/components/Tag/Tag.tsx:21]: And just pass it here:

```
    <div style={marginStyles} className={cls}>
```

**Comment by alexpricedev** [comment]
Can you also upload a new screenshot pls :)
_We can sort the colours when it's all merged_

**Pull request merged**